src/lib/ledger_catchup/catchup_worker.ml

```diff
@@ -0,0 +1,57 @@
+open Core
+open Async
+open Coda_base
+
+let dispatch rpc query peer =
```

Comment:
I think with Corey's new networking abstraction, we can just setup a pipe that we write to in order to dispatch network requests. I think this is preferable moving forward (if it's ready to go), but let's chat about this.

Reply:
For RPCs where we need a response, pipes are the wrong interface because then we need a connection manager to associate outgoing and incoming data, with some connection state that peers can't spoof. This ends up turning into a table of `(IP, seqno) -> response Ivar.t` -- basically pulling the implicit data flow of RPC into an explicit table. The right interface for that is `request -> response Deferred.Or_error.t`, which happens to be interface that RPC already has!

We shouldn't have this worker module. For now, this should just be a part of Coda_networking.ml. That will still ultimately contain all the RPC setup/listen code (post refactor), it will just take a list of `Rpc.implementation` instead of callbacks.